provisioning/miniprov/hare_mp/main.py

```diff
@@ -280,36 +280,43 @@ def test_IVT(args):
 def reset(args):
     try:
         rc = 0
-        util: ConsulUtil = ConsulUtil()
-
-        if is_cluster_running():
-            logging.info('Cluster is running, shutting down')
-            shutdown_cluster()
-
-        keys: List[KeyDelete] = [
-            KeyDelete(name='epoch', recurse=False),
-            KeyDelete(name='eq-epoch', recurse=False),
-            KeyDelete(name='last_fidk', recurse=False),
-            KeyDelete(name='leader', recurse=False),
-            KeyDelete(name='m0conf/', recurse=True),
-            KeyDelete(name='processes/', recurse=True),
-            KeyDelete(name='stats/', recurse=True)
-        ]
-
-        logging.info('Deleting Hare KV entries (%s)', keys)
-        if not util.kv.kv_delete_in_transaction(keys):
-            logging.error('Error during key delete in transaction')
-            rc = -1
 
         exit(rc)
     except Exception as error:
         logging.error('Error during reset (%s)', error)
         exit(-1)
 
 
+def kv_cleanup():
+    rc = 0
+    util: ConsulUtil = ConsulUtil()
+
+    if is_cluster_running():
+        logging.info('Cluster is running, shutting down')
+        shutdown_cluster()
+
+    keys: List[KeyDelete] = [
+        KeyDelete(name='epoch', recurse=False),
+        KeyDelete(name='eq-epoch', recurse=False),
+        KeyDelete(name='last_fidk', recurse=False),
+        KeyDelete(name='leader', recurse=False),
+        KeyDelete(name='m0conf/', recurse=True),
+        KeyDelete(name='processes/', recurse=True),
+        KeyDelete(name='stats/', recurse=True)
+    ]
+
+    logging.info('Deleting Hare KV entries (%s)', keys)
+    if not util.kv.kv_delete_in_transaction(keys):
+        logging.error('Error during key delete in transaction')
+        rc = -1
```

Comment:
Please raise exception instead of using C-style return codes. This will make code shorter and there will be less chance to swallow an error accidentally.

Reply:
Added new changes into that.